networking/src/JsonLog/JsonLogT.hs

```diff
@@ -149,7 +149,7 @@ jsonLogDefault jlc x =
                 `catchAny` \e -> do
                     logWarning $ sformat ("error in deciding whether to json log: "%shown) e
                     return False
-            when b $ liftIO (withMVar v $ flip hPut $ encode event)
+            when b $ liftIO (withMVar v $ \h -> (hPut h (encode event) >> hFlush h))
```

Comment:
Wouldn't it be better to set the buffering mode of the handle appropriately, instead of manually flushing after each log entry?

Reply:
Speaking of which, it seems the buffering mode is set to `NoBuffering`, which seems a bit extreme (see lib/src/Pos/Launcher/Resource.hs)

Reply:
I have seen that `NoBuffering` is set but I added `flush`, which is redundant in `NoBuffering` mode, in the case of somebody in the future changes it forgetting to add it.

Reply:
But isn't `NoBuffering` too aggressive? I think it would be preferable to have `LineBuffering`, and use an `hPutLn`, which would also make the file much easier to load up in an editor and have a glance at it.